### PR TITLE
Add "approved for" to a PR's state

### PR DESCRIPTION
**Pull request by maartenberg**
This PR alters Hoff to include a "approval type" in a PR's state.
The current means of approving a PR (`<prefix> merge`) always mark a PR to be merged, but we are planning to add another approval command (`<prefix> merge and deploy`) to trigger different behaviour if the resulting merge succeeds.

This PR:
 - Defines a new enumeration `ApprovedFor` that for now contains just one option `Merge`,
 - Defines a type alias `Approval = Approval { approver :: Username, approvedFor :: ApprovedFor }`, to model that a user approves a PR with a specific approval type,
 - Replaces `PullRequest`'s `approvedBy :: Maybe Username` with `approval :: Maybe Approval`,
 - Alters `isMergeCommand` to return `Maybe ApprovedFor` depending on the merge command used,
 - Alters the various sites where `approvedBy` was used to use `approval` instead, and
 - Updates the tests to use `approval` instead of `approvedBy`.

### CODE REVIEW

File: src/Project.hs
Comment (rkrzr) [inline src/Project.hs:101]: s/"just merge"/just "merge"

File: src/Project.hs
Comment (rkrzr) [inline src/Project.hs:102]: I would make this an ADT instead of a `type` for extra type safety.

File: src/Logic.hs
Comment (rkrzr) [inline src/Logic.hs:283]: s/isMergeCommand/parseMergeCommand (?)

**Review by rkrzr — COMMENTED** [review]
I would change the `Approval` type. The rest looks good.

File: src/Project.hs
Comment (maartenberg) [reply src/Project.hs:102]: So something like:

```haskell
data Approval =
  { approvedBy :: Username
  , approvedFor :: ApprovedFor
  }
```

Then?

Comment (rkrzr) [reply src/Project.hs:102]: Yes, exactly.

Comment (maartenberg) [reply src/Project.hs]: Done. I renamed the `approvedBy` field to `approver` to avoid shadowing as there are some places where `approvedBy` is used as a variable name.

**Review by rkrzr — APPROVED** [review]
LGTM!

**Comment by maartenberg** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved by @maartenberg, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as 333a1938a048994b90f2353e4b273202608bed1b, waiting for CI …

**Pull request merged**
